libr/core/cmd_search.c

```diff
@@ -1060,19 +1061,23 @@ static RList *construct_rop_gadget(RCore *core, ut64 addr, ut8 *buf, int buflen,
 	}
 	while (nb_instr < max_instr) {
 		ht_uu_insert (localbadstart, idx, 1);
-		r_asm_set_pc (core->assembler, addr);
-		if (!r_asm_disassemble (core->assembler, &asmop, buf + idx, buflen - idx)) {
-			goto ret;
-		}
-		RAnalOp aop;
-		r_anal_op (core->anal, &aop, addr, buf + idx, buflen - idx, R_ANAL_OP_MASK_BASIC);
+		r_anal_op (core->anal, &aop, addr, buf + idx, buflen - idx, R_ANAL_OP_MASK_DISASM);
+
 		if (nb_instr == 0 && is_end_gadget (&aop, 0)) {
 			valid = false;
 			goto ret;
 		}
-		const int opsz = asmop.size;
+		const int opsz = aop.size;
 		// opsz = r_strbuf_length (asmop.buf);
-		char *opst = r_asm_op_get_asm (&asmop);
+		char *opst = aop.mnemonic;
+		if (!opst) {
+			r_asm_set_pc (core->assembler, addr);
+			if (!r_asm_disassemble (core->assembler, &asmop, buf + idx, buflen - idx)) {
```

Comment:
Keep asmop local here

Reply:
Can we have a warning once for those analop archs that dont provide/handle disasm instead? This is a bug we should handle in the plugins not as a workaround here. Soon or late rasm api should be using anal plugs